Feature/edit skill types

Nizar-Kaouri opened the pull request.

Nizar-Kaouri committed:
Duplicate button created

Nizar-Kaouri committed:
SkillTypes View created
SkillTypes Edit button created
Skilltypes Delete button created
Skilltypes Create Button created
SkillTypes in Scenario Simulation "GET"-Method modified

Nizar-Kaouri committed:
SkillTypes View created
SkillTypes Edit button created
Skilltypes Delete button created
Skilltypes Create Button created
SkillTypes in Scenario Simulation "GET"-Method modified
Small Modification in User Manager

Nizar-Kaouri committed:
Input Fields conditions added

Nizar-Kaouri committed:
SkillType GET-Method fixed

Nizar-Kaouri committed:
Restrictions and Descriptions to the Editors added

Nizar-Kaouri committed:
Restrictions and Descriptions to the Editors added

Nizar-Kaouri committed:
SkillType-Names handling and Restrictions created

Nizar-Kaouri committed:
Navbar anpassen

AmineAmzil commented on backend/custom_user/migrations/0002_alter_user_managers.py:
try to remove this file

AmineAmzil commented on frontend/src/pages/SkillTypesOverview.jsx:
```diff
@@ -0,0 +1,726 @@
+      setSkilltypes(skilltypesData.data);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(error);
```
no console.log

AmineAmzil commented on frontend/src/pages/SkillTypesOverview.jsx:
```diff
@@ -0,0 +1,726 @@
+      }
+    }
+  } catch (error) {
+    console.log(error);
```
s.o.

AmineAmzil commented on frontend/src/pages/SkillTypesOverview.jsx:
```diff
@@ -0,0 +1,726 @@
+      const nameExists = skilltypes.some((skill) => {
+
+  return (
+    skill.name &&
```
fomatierung

AmineAmzil commented on frontend/src/pages/SkillTypesOverview.jsx:
```diff
@@ -0,0 +1,726 @@
+        },
+      });
+      await res.json();
+      await fetchSkillTypes();
```
Refresh only when res.ok()

AmineAmzil commented on frontend/src/pages/SkillTypesOverview.jsx:
```diff
@@ -0,0 +1,726 @@
+      const updatedSkillType = await res.json();
+
+      // Check if the name already exists
+      const nameExists = skilltypes.some((skill) => {
```
this should be first. Check if the name is unique, if unique then send the request, if the request ok is, then refresh the page

AmineAmzil commented on frontend/src/pages/SkillTypesOverview.jsx:
Formatting !!

Nizar-Kaouri committed:
fixed Code Navbar so that only Creator/Admin can see Settings
fixed Update Method so that the name gets checked first
fixed Delete Method so that it only loads the page again if it was successful
remove Console.log()

Nizar-Kaouri committed:
Name Checking Method is now an independent Method that can be called when needed

Nizar-Kaouri committed:
changed Descriptions
small bug fix in checking name method

Nizar-Kaouri committed:
Formatting

Nizar-Kaouri committed:
reverted team.py

Nizar-Kaouri committed:
Formatting
